File: frontend/src/types/repositories.ts
export interface IngestRequest {
  repository_url: string
}

export interface DocumentMetadata {
  repository: string
  file_path: string
  file_name: string
  language: string
  size: number
  content_preview: string
}

export interface IngestResponse {
  repository: string
  files_processed: number
  files_skipped: number
  chunks_created: number
  embeddings_created: number
  documents: DocumentMetadata[]
}

File: backend/app/schemas/repositories.py
from typing import Annotated

from pydantic import BaseModel, Field, StringConstraints

RepositoryURL = Annotated[
    str,
    StringConstraints(strip_whitespace=True, min_length=1, max_length=2048),
]


class IngestRequest(BaseModel):
    repository_url: RepositoryURL


class DocumentResponse(BaseModel):
    repository: str
    file_path: str
    file_name: str
    language: str
    size: int
    content_preview: str


class IngestResponse(BaseModel):
    repository: str
    files_processed: int
    files_skipped: int
    chunks_created: int = 0
    documents: list[DocumentResponse] = Field(default_factory=list)
